File: 832_flipping_an_image/flipping_an_image.py
from typing import *

class Solution:
    def flipAndInvertImage(self, A: List[List[int]]) -> List[List[int]]:
        for i in range(len(A)):
            left, right = 0, len(A[i]) - 1
            while left < right:
                A[i][left], A[i][right] = A[i][right] ^ 1, A[i][left] ^ 1
                left += 1
                right -= 1
            if left == right:
                A[i][left] ^= 1
        return A

File: 832_flipping_an_image/flipping_an_image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] A) {
        int n = A.length;
        for (int i = 0; i< n ; i++) {
            int left=0, right=A[i].length - 1;
            while (left < right) {
                int tmp = A[i][left];
                A[i][left] = A[i][right] ^ 1;
                A[i][right] = tmp ^ 1;
                left++;
                right--;
            }
            if (left == right) {
                A[i][left] ^= 1;
            }
        }
        return A;
    }
}